Sources/Fuzzilli/Core/CodeGenerators.swift

```diff
@@ -573,8 +573,21 @@ public let CodeGenerators: [CodeGenerator] = [
     },
 
     CodeGenerator("ForOfLoopGenerator", input: .iterable) { b, obj in
-        b.forOfLoop(obj) { _ in
-            b.generateRecursive()
+        var indices: [Int] = []
```

Comment:
I suspect that the ForOfLoop with destructuring has a pretty high failure rate, since stuff like
```javascript
var a = [1,2,3]
for ([x, y] of a) { console.log(x, y); }
// Uncaught:
// TypeError: number 1 is not iterable (cannot read property Symbol(Symbol.iterator))
```
throws an exception.

So I think there should at least be two separate CodeGenerators, and the one that does destructuring probably shouldn't run in `.conservative` mode at all (maybe with a comment "until we can track array element types").

I'm not sure if having two separate FuzzIL Operations would make sense as well. Maybe it does: if the OperationMutator is able to mutate the `indices` of a `ForOfLoop` (it probably should?), then for a "regular" for-of loop (one with `indices = [0]`), it would most of the time construct something that's invalid (i.e. `for ([,,,x] of y)`, where `y` is a normal array, not a nested one). Further, it probably would make lifting a tiny bit easier (get's rid of `if op.indices.count > 1 {` below). So I think I'm in favour of adding a separate `ForOfLoopWithDestruct` operation. WDYT?

Reply:
Yeah this makes sense and I agree, I'll add a new op,